pkg/services/ngalert/database.go

```diff
@@ -0,0 +1,120 @@
+package ngalert
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+func getAlertDefinitionByID(alertDefinitionID int64, sess *sqlstore.DBSession) (*AlertDefinition, error) {
+	alertDefinition := AlertDefinition{}
+	has, err := sess.ID(alertDefinitionID).Get(&alertDefinition)
+	if !has {
+		return nil, ErrAlertDefinitionNotFound
+	}
+	if err != nil {
+		return nil, err
+	}
+	return &alertDefinition, nil
+}
+
+// DeleteAlertDefinitionByID is a handler for deleting an alert definition.
+// It returns models.ErrAlertDefinitionNotFound if no alert definition is found for the provided ID.
+func (ng *AlertNG) DeleteAlertDefinitionByID(query *DeleteAlertDefinitionByIDQuery) error {
+	return ng.SQLStore.WithTransactionalDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		res, err := sess.Exec("DELETE FROM alert_definition WHERE id = ?", query.ID)
+		if err != nil {
+			return err
+		}
+
+		rowsAffected, err := res.RowsAffected()
+		if err != nil {
+			return err
+		}
+		query.RowsAffected = rowsAffected
+		return nil
+	})
+}
+
+// GetAlertDefinitionByID is a handler for retrieving an alert definition from that database by its ID.
+// It returns models.ErrAlertDefinitionNotFound if no alert definition is found for the provided ID.
+func (ng *AlertNG) GetAlertDefinitionByID(query *GetAlertDefinitionByIDQuery) error {
+	return ng.SQLStore.WithTransactionalDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		alertDefinition, err := getAlertDefinitionByID(query.ID, sess)
+		if err != nil {
+			return err
+		}
+		query.Result = alertDefinition
+		return nil
+	})
+}
+
+// SaveAlertDefinition is a handler for saving a new alert definition.
+func (ng *AlertNG) SaveAlertDefinition(cmd *SaveAlertDefinitionCommand) error {
```

Comment:
Doesn't have to be public:
```suggestion
// saveAlertDefinition is a handler for saving a new alert definition.
func (ng *AlertNG) saveAlertDefinition(cmd *SaveAlertDefinitionCommand) error {
```